include/picongpu/particles/atomicPhysics/stateRepresentation/ConfigNumber.hpp

```diff
@@ -0,0 +1,350 @@
+ * 3 |0    |1    |0
+ * 4 |1    |1    |0
+ * 5 |2    |1    |0
+ * 6 |0    |2    |0
+ * 7 |1    |2    |0
+ * 8 |2    |2    |0
+ * 9 |0    |3    |0
+ * ...
+ *
+ * analytical formula:
+ * # ... configNumber assigned
+ * # = N_1 *(g(0) + 1)+ N_2 * (g(1)+1) + N_3 * (g(2)+1) * (g(1) + 1)) + ...
+ * # = Sum_{n=1}^{n_max}[ N_n * Produkt_{i=1}^{n} (g(i-1) + 1) ]
+ *
+ * g(n) ... maximum number of electrons in a given level n
+ * g(n) = 2 * n^2
+ * quick reference:
+ * https://en.wikipedia.org/wiki/Electron_shell#Number_of_electrons_in_each_shell
+ *
+ * Note: a superconfiguration only stores occupation numbers, NOT spin or
+ *  angular momentumm. This was done due to memory constraints.
+ *
+ * further information see:
+ *  https://en.wikipedia.org/wiki/Hydrogen-like_atom#Schr%C3%B6dinger_solution
+ *
+ * @todo 2020-07-01 BrianMarre: implement usage of current charge to account for
+ * one more level than actually saved, => n_max effective = n_max + 1
+ */
+
+#pragma once
+
+
+#include <pmacc/algorithms/math/defines/comparison.hpp>
+#include <pmacc/assert.hpp>
+#include <pmacc/math/Vector.hpp>
+
+#include <cstdint>
+
+namespace picongpu
+{
+    namespace particles
+    {
+        namespace atomicPhysics
+        {
+            namespace stateRepresentation
+            {
+                /** This class implements the actual storage of the superconfigurations by index
+                 *
+                 * @tparam T_DataType ... unsigned integer data type to represent the config
+                 *                              number
+                 * @tparam T_numberLevels ... max principle quantum number, n_max, represented in
+                 *                              this confignumber
+                 * @tparam T_atomicNumber ... atomic number of the ion, in units of elementary charge
+                 *
+                 * private member:
+                 *   configNumber    ... actual stored configNumber ID
+                 *
+                 * public members:
+                 *   DataType        ... access to T_DataType
+                 *   numberLevels    ... acess to T_numberLevels
+                 *
+                 * private methods:
+                 *   uint g( uint n )
+                 *       g(n) = n^2 maximum possible occupation number of the n-th shell
+                 *   uint numberOfOccupationNumberValuesInShell( uint n )
+                 *       number of different occupation number values possible for the n-th shell
+                 *   T_DataType stepLength( uint n)
+                 *       step length, number of table entries per occupation number VALUE of
+                 *       the current principal quantum number n, of the n-th shell
+                 *   void nextStepLength( T_DataType* currentStepLength, uint current_n )
+                 *       returns current step length based on previous
+                 *
+                 * public methods:
+                 *   T_DataType numberStates()
```

Comment:
The function documentation should be part of the function description else doxygen will not show it.

Reply:
they are documented both at the function itself and here.
Intention was to provide a short overview.

Reply:
done